portfolio/src/main/java/com/google/sps/helper/Pagination.java

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.helper;
+
+import com.google.sps.helper.ValidationResult;
+import com.google.sps.servlets.DataServlet;
+import javax.servlet.http.HttpServletRequest;
+
+public final class Pagination {
+
+  public static final String PAGE_KEY = "page";
+  public static final String NUM_PER_PAGE_KEY = "numPerPage";
+
+  private static final int DEFAULT_NUM_PER_PAGE = 10;
+
+  /**
+   * Get and validate pagination options from an incoming request.
+   * 
+   * If either the page number parameter or the number per page parameter
+   * are not supplied, they are supplemented with default values.
+   * @return Returns a pagination object or a validation error message.
+   */
+  public static ValidationResult<Pagination> getIncomingPagination(
+      HttpServletRequest request) {
+    String pageNumParam = DataServlet.getParameter(request, PAGE_KEY, null);
+    String numPerPageParam = DataServlet.getParameter(request, NUM_PER_PAGE_KEY, null);
+
+    String validationString = "";
+
+    Integer pageNum = tryParse(pageNumParam, 0);
+    Integer numPerPage = tryParse(numPerPageParam, DEFAULT_NUM_PER_PAGE);
+
+    if (pageNum == null) {
+      validationString += "Page number is not a valid number.";
+    }
+    if (numPerPage == null) {
+      if (validationString.length() > 0) {
+        validationString += " ";
+      }
```

Comment:
```suggestion
    if (pageNum == null) {
      validationString += "Page number is not a valid number. ";
    }
    if (numPerPage == null) {
```

Reply:
Okay I think I know what went wrong in your first suggestion. You might've forgotten to add in the space on the first one! Do you think this is more stylistic? I did it to avoid having trailing whitespace if only the first was added.. Is this a common solution that you encounter?

Reply:
Oops, you're probably right. The common solution is actually to put all the strings into some sort of list and join them on " " I think :)

Reply:
Actually that does seem like a prettier solution. Using a join function. I'm going to change them to that!